lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -28,6 +28,51 @@ using namespace circt;
 using namespace circt::calyx;
 using namespace mlir;
 
+//===----------------------------------------------------------------------===//
+// Utilities related to Direction
+//===----------------------------------------------------------------------===//
+
+Direction direction::get(bool a) { return static_cast<Direction>(a); }
+
+SmallVector<Direction> direction::genInOutDirections(size_t nIns,
+                                                     size_t nOuts) {
+  SmallVector<Direction> dirs;
+  std::generate_n(std::back_inserter(dirs), nIns,
+                  [] { return Direction::Input; });
+  std::generate_n(std::back_inserter(dirs), nOuts,
+                  [] { return Direction::Output; });
+  return dirs;
+}
+
+IntegerAttr direction::packAttribute(ArrayRef<Direction> directions,
+                                     MLIRContext *ctx) {
+  const size_t size = directions.size();
```

Comment:
`const` not necessary.
Tiny nit: please use more descriptive name such as `numDirections`.